### ISSUE TITLE
Support 3.11 on aarch64

### ISSUE THREAD

**Issue by gnossen** [grpc/grpc#30927]
The latest version of dockcross aarch64 does not yet support Python 3.11, blocking us from creating these artifacts. Once they _do_ support it, we should start producing these artifacts. This image has not been updated on dockerhub for a year, so it's possible that this may require more than just waiting for a new release. I've raised [the issue here](https://github.com/dockcross/dockcross/issues/722) to find out.

**Comment by mgu** [comment]
maybe an alternative could be to switch to pypa/cibuildwheel ?

**Comment by samanta-widjaja** [comment]
is there a solution. i can't install due the same problem too

**Comment by mgu** [comment]
@gnossen do you think I could open a PR to switch from dockcross to qemu+aarch64 image instead ?
Even if it is super slow, at least it would work

**Comment by gnossen** [comment]
> **maybe an alternative could be to switch to pypa/cibuildwheel ?**

I haven't looked deeply into this, but it seems that integrates at the level of the CI system, e.g. Github Actions, Travis, Circle CI. For software supply chain security reasons, we use a private system called Kokoro and we would not be able to use these cibuildwheel directly, though we might be able to pull some components out of it to enable this case.

> do you think I could open a PR to switch from dockcross to qemu+aarch64 image instead ?

We'd have to play around with CI configuration to make sure that our PR presubmits don't get blocked by a 5 hour build, but in principle, yes, a contribution like that would be greatly appreciated. Thank you for the offer!

**Comment by mgu** [comment]
I think the qemu is way too slow (the build timed out after 2h 😓 ).
I'll do some more tests to understand what can be done

**Comment by mgu** [comment]
```
PASSED: build_artifact.python_manylinux2014_aarch64_cp311-cp311 [time=798.4sec, retries=0:0]
```
I've been able to build a dockcross image with python 3.11 🎉 
I'll try to create PRs for dockross & grpc in the next few days

**Comment by kasium** [comment]
It seems that the latest release 1.15.3 still doesn't include a 3.11 aarch wheel. Can you please check?

**Comment by mgu** [comment]
Hi @kasium 
From my understanding, the aarch wheel will be available from 1.53.x (1.51.3 is a maintenance release and 1.52 was yanked)

**Comment by kasium** [comment]
@mgu I see. Thanks!
